client/client_test.go

```diff
@@ -1186,6 +1156,37 @@ func testPublishNoData(t *testing.T, rootType string, serverManagesSnapshot bool
 	}
 }
 
+// Publishing an uninitialized repo will fail, but initializing and republishing
+// after should succeed
+func TestPublishUninitializedRepo(t *testing.T) {
+	gun := "docker.com/notary"
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	// uninitialized repo should fail to publish
+	tempBaseDir, err := ioutil.TempDir("", "notary-tests")
+	assert.NoError(t, err)
+	defer os.RemoveAll(tempBaseDir)
+
+	repo, err := NewNotaryRepository(tempBaseDir, gun, ts.URL,
+		http.DefaultTransport, passphraseRetriever)
+	assert.NoError(t, err, "error creating repository: %s", err)
+	err = repo.Publish()
+	assert.Error(t, err)
+
+	// no metadata created
+	assertRepoHasExpectedMetadata(t, repo, data.CanonicalRootRole, false)
+	assertRepoHasExpectedMetadata(t, repo, data.CanonicalSnapshotRole, false)
+	assertRepoHasExpectedMetadata(t, repo, data.CanonicalTargetsRole, false)
+
+	// now, initialize and republish in the same directory
+	rootPubKey, err := repo.CryptoService.Create("root", data.ECDSAKey)
+	assert.NoError(t, err, "error generating root key: %s", err)
+
+	assert.NoError(t, repo.Initialize(rootPubKey.ID()))
```

Comment:
That is the assumption, yes :)  There is a test here https://github.com/docker/notary/pull/384/files#diff-fce84029d346504e44323c30d21a3808R402 that tests that initializing a repo creates the right metadata, keys, and certs.  But I can also make the same assertions here.